nvidia_gpu_stats/cmd/nvidia_gpu_stats/main.go

```diff
@@ -0,0 +1,241 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/shirou/gopsutil/v4/process"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+type GPUNvidia struct {
+	pid      int
+	nvmlInit nvml.Return
+}
+
+func NewGPUNvidia(pid int) *GPUNvidia {
+	g := &GPUNvidia{pid: pid}
+	return g
+}
+
+func (g *GPUNvidia) gpuInUseByProcess(device nvml.Device) bool {
+	proc, err := process.NewProcess(int32(g.pid))
+	if err != nil {
+		return false
+	}
+
+	ourPids := make(map[int32]struct{})
+	ourPids[int32(g.pid)] = struct{}{}
+
+	childProcs, err := proc.Children()
+	if err == nil {
```

Comment:
leave a comment that you are using `err==nil` as it is not the standard pattern.
see here: https://google.github.io/styleguide/go/best-practices#signal-boosting